HLSPlugin/src/com/kaltura/hls/manifest/HLSManifestEncryptionKey.as

```diff
@@ -106,18 +107,76 @@ package com.kaltura.hls.manifest
 		 * Decrypts a video or audio stream using AES-128 with the provided initialization vector.
 		 */
 		
-		public function decrypt( data:ByteArray, iv:ByteArray ):void
+		public function decrypt( data:ByteArray, iv:ByteArray ):ByteArray
 		{
 			trace("got " + data.length + " bytes");
 			if(data.length == 0)
-				return;
+				return data;
 				
 			var startTime:uint = getTimer();
-			decryptAES( data, _keyData, iv );
-			if ( usePadding ) unpad( data );
+			_key = new FastAESKey(_keyData);
+			iv.position = 0;
+			iv0 = iv.readUnsignedInt();
+			iv1 = iv.readUnsignedInt();
+			iv2 = iv.readUnsignedInt();
+			iv3 = iv.readUnsignedInt();
+			data.position = 0;
+			data = _decryptCBC(data,data.length);
+			//decryptAES( data, _keyData, iv );
+			if ( usePadding ){
+				data = unpad( data );
+			}
 			trace( "DECRYPTION OF " + data.length + " BYTES TOOK " + ( getTimer() - startTime ) + " MS" );
```

Comment:
I would probably modify this log line to only fire if exceeding a threshold rather than all the time.